src/main/kotlin/no/nav/familie/tilbake/dokumentbestilling/felles/task/PubliserJournalpostTask.kt

```diff
@@ -40,13 +42,27 @@ class PubliserJournalpostTask(
         val journalpostId = task.metadata.getProperty("journalpostId")
         val behandlingId = UUID.fromString(task.payload)
 
-        prøvDistribuerJournalpost(journalpostId, task, behandlingId)
-
         if (featureToggleService.isEnabled(DSITRIBUER_TIL_MANUELLE_BREVMOTTAKERE)) {
-            val manuelleAdresser = manuellBrevmottakerService.hentBrevmottakereAsManuellAdresse(behandlingId)
-            manuelleAdresser.forEach { manuellAdresse ->
+            val brevmottakere = manuellBrevmottakerService.hentBrevmottakere(behandlingId)
+
+            val dødsboAdresser = brevmottakere.filter { it.type == MottakerType.DØDSBO }
+            val utenlandskeAdresser = brevmottakere.filter { it.type == MottakerType.BRUKER_MED_UTENLANDSK_ADRESSE }
+
+            if (dødsboAdresser.isEmpty() && utenlandskeAdresser.isEmpty()) {
+                prøvDistribuerJournalpost(journalpostId, task, behandlingId)
+            }
+
+            val manuelleAddresser = when {
+                dødsboAdresser.isNotEmpty() -> dødsboAdresser.toList()
+                utenlandskeAdresser.isNotEmpty() -> brevmottakere
```

Comment:
Trenger egentlig ikke denne when-branchen for utenlandsk adresse, siden else-branchen returnerer det samme 
```suggestion
```